### PR TITLE
First Task

### PR DESCRIPTION
**Pull request by ahmeddmujtaba**
Some work was done to convert from Python2.7 to Python3 and PyTorch 0.1.12 to PyTorch 2.0.0.
The model was trained for 30,000 epochs and tested on Python 3.11.5 and PyTorch 2.0.0 and then saved.

The scipy.misc.imsave function is deprecated. Using imageio package instead

There's also a new "test" function in cyclegan_arch/mnist_to_svhn/solver.py. This function is similar to the train function but does not do any gradient updates on the model and prints out average loss. 

Screen Shot 2024-01-03 at 1 33 53 PM

Sample Sample SVHN to MNIST conversion after 500 epochs:
image

Sample SVHN to MNIST conversion after 30,000 epochs:
image

### CODE REVIEW

File: cyclegan_arch/mnist_to_svhn/data_loader.py [hunk 1]
```diff
@@ -6,9 +6,9 @@ def get_loader(config):
     """Builds and returns Dataloader for MNIST and SVHN dataset."""
     
     transform = transforms.Compose([
-                    transforms.Scale(config.image_size),
+                    transforms.Resize(config.image_size),
                     transforms.ToTensor(),
-                    transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
```
Comment (ahmeddmujtaba) [inline cyclegan_arch/mnist_to_svhn/data_loader.py:11]: Throws error when using this transform on MNIST which is grayscale and this function expected RGB.
Changed to handle both RGB and Grayscale

File: cyclegan_arch/mnist_to_svhn/solver.py [hunk 2]
```diff
@@ -231,6 +235,160 @@ def set_expectation_and_std(self):
 
     def as_np(self, data):
         return data.cpu().data.numpy()
+
```
Comment (amujtaba00) [inline cyclegan_arch/mnist_to_svhn/solver.py:238]: The new test function.
Copied mostly from the train function. 
No grad updates and uses test loader.
Calculates all the same losses.

File: cyclegan_arch/mnist_to_svhn/solver.py [hunk 3]
```diff
@@ -231,6 +235,160 @@ def set_expectation_and_std(self):
 
     def as_np(self, data):
         return data.cpu().data.numpy()
+    
+    def test(self,svhn_test_loader,mnist_test_loader):
+        svhn_test_iter = iter(svhn_test_loader)
```
Comment (ahmeddmujtaba) [inline cyclegan_arch/mnist_to_svhn/solver.py:240]: The new test function.
Copied mostly from the train function.
No grad updates and uses test loaders.
Calculates all the same losses.
